src/oop/user_defined_classes/hard/q2/MainTest.java

```diff
@@ -0,0 +1,120 @@
+package oop.user_defined_classes.hard.q2;
+
+import global.ObjectTest;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+    private final String testClassName = "DishWasher";
+    private final String intAttributeName1 = "forks";
+    private final String intAttributeName2 = "knives";
+    private final String methodName = "loadDishWasher";
+    private ObjectTest classInstance;
+
+    @BeforeEach
+    public void setUp() {
+        String testClassString = "oop.user_defined_classes.hard.q2." + testClassName;
+        classInstance = new ObjectTest(testClassString);
+    }
+
+    @Test
+    public void dishWasherClassHasCorrectAttributes() {
+        String missingAttributeMessage =
+                "The attribute %s could not be found in your class. Please make sure you have added it and that it is spelled correctly.";
+        String wrongTypeMessage = "The attribute %s does not have the correct type.";
```

Comment:
Can we say "The x attribute ..." instead of  "The attribute x ..." ? I feel like it sounds more grammatically correct.